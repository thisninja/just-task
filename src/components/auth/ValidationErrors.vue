<template>
  <div class="validation-errors">
    <b>
      {{ INFO_MESSAGE_TEXT }}:
    </b>
    <ul>
      <transition-group name="slide" mode="out-in">
        <li
        class="validation-errors__error"
        v-for="(error, index) in errors"
        :key="index"
        >
          {{ error }}
        </li>
      </transition-group>
    </ul>
  </div>
</template>

<script>
import {
  INFO_MESSAGE_TEXT,
} from './constants';

export default {
  data() {
    return {
      INFO_MESSAGE_TEXT,
    }
  },
  props: {
    errors: {
      type: Array,
      required: true,
    }
  }
}
</script>

<style lang="scss" scoped>
@import "@/assets/scss/_variables.scss";

.validation-errors {
  display: inline-block;

  &__error {
    color: $material-red-200;
    text-align: left;
  }
}
</style>
