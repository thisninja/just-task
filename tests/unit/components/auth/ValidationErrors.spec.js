import { shallowMount } from '@vue/test-utils';
import ValidationErrors from '@/components/auth/ValidationErrors.vue';

describe('ValidationErrors.vue', () => {
  const errors = [
    'foo',
    'bar'
  ];

  it('should match snapshot', () => {
    const wrapper = shallowMount(ValidationErrors, {
      propsData: {
        errors,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
