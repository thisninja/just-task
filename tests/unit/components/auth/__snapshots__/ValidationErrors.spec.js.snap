// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationErrors.vue should match snapshot 1`] = `
<div class="validation-errors"><b>
    Please correct the following error(s):
  </b>
  <ul><span name="slide" mode="out-in"><li class="validation-errors__error">
        foo
      </li><li class="validation-errors__error">
        bar
      </li></span></ul>
</div>
`;
